features/network_failures.feature

```diff
@@ -0,0 +1,92 @@
+@uses.config.contract_token
+Feature: Ensure network errors are handled gracefully across various services
+
+  Scenario Outline: Various HTTP errors are handled gracefully on attaching contract token
+    # This test simulates various HTTP errors by mocking the response from the serviceclient
+    # when trying to attach contract token
+    Given a `<release>` `<machine_type>` machine with ubuntu-advantage-tools installed
+    When I create the file `/tmp/response-overlay.json` with the following:
+      """
+      {
+        "https://contracts.canonical.com/v1/context/machines/token": [
+          {
+            "code": <response_code>,
+            "response": {
+              "error": <error_message>
+            }
+          }
+        ]
+      }
+      """
+    And I append the following on uaclient config:
+      """
+      features:
+      serviceclient_url_responses: "/tmp/response-overlay.json"
```

Comment:
```suggestion
      features:
        serviceclient_url_responses: "/tmp/response-overlay.json"
```

Small indent issue causes test to fail, never sets the overlay and the machine ends up attached

Reply:
😭😭 good catch. Thanks. (I didn't actually know indentation mattered like that)

Reply:
@a-dubs this is a yaml file, the responses key is nested under features:

Reply:
oh wait i was being dumb yeah it is a yaml file. i thought the `.feature` file was throwing an error due to indentation lol.